bin/node/cli/src/cli.rs

```diff
@@ -50,32 +52,11 @@ impl GetSharedParams for CustomSubcommands {
 /// Please note: this command currently only works on an empty database!
 #[derive(Debug, StructOpt, Clone)]
 pub struct FactoryCmd {
-	/// How often to repeat. This option only has an effect in mode `MasterToNToM`.
-	#[structopt(long="rounds", default_value = "1")]
-	pub rounds: u64,
-
-	/// MasterToN: Manufacture `num` transactions from the master account
-	///            to `num` randomly created accounts, one each.
-	///
-	/// MasterTo1: Manufacture `num` transactions from the master account
-	///            to exactly one other randomly created account.
-	///
-	/// MasterToNToM: Manufacture `num` transactions from the master account
-	///               to `num` randomly created accounts.
-	///               From each of these randomly created accounts manufacture
-	///               a transaction to another randomly created account.
-	///               Repeat this `rounds` times. If `rounds` = 1 the behavior
-	///               is the same as `MasterToN`.{n}
-	///               A -> B, A -> C, A -> D, ... x `num`{n}
-	///               B -> E, C -> F, D -> G, ...{n}
-	///               ... x `rounds`
-	///
-	/// These three modes control manufacturing.
-	#[structopt(long="mode", default_value = "MasterToN")]
-	pub mode: node_transaction_factory::Mode,
-
-	/// Number of transactions to generate. In mode `MasterNToNToM` this is
-	/// the number of transactions per round.
+	/// Transaction name.
+	#[structopt(long="tx-name", default_value = "transfer")]
```

Comment:
is there a module selected as well? or they will all be chosen from balances? Can I call remark here?

Reply:
That's old, going to be fixed today. 

In general, this is just a draft. I'm exploring what can be done and if it's useful.

My current idea is to have:
- Simple command like `substrate bench` that executes all transactions (using default values) of all modules, N number of times and make a report. 
- Same for modules and specific transactions, i.e. `bench staking` (all tx in staking), `bench staking bond` (just execute bond N times).
- I found that many times we want to loop and add dependencies between tx, for that I would use some kind of automata and pass the file. So it will be `bench staking.txt`, with edges within the file as: `0 1 staking_unbond, 1 0 staking_rebond`, (`src_state dest_state tx`).
- Something I didn't figure out is how to specify the parameters of each tx. I was thinking to have a name (string) for a specific instance (value) of a type. Some edge like `0 1 balances_transfer alice 10`, could fetch the values like `Address::test_value("alice"), Balances::test_value("10")`.

Also, I'm not sure about the whole automaton idea, may be a waste of time. But it's nice to not need to recompile everything when you benchmark do something wrong :).

Reply:
@shawntabrizi  @mattrutherford  I think you guys might want to align together?